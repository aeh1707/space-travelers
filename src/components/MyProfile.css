.my-profile-container {
  width: 70vw;
  margin: 0 auto;
} 

.myprofile {
  display: flex;
  gap: 25px;
}

.joined-missions,
.reserved-rockets {
  width: 30vw;
}

.myprofile ul {
  list-style: none;
  padding: 0;
}

.myprofile li {
  padding: 20px;
  border: 1px solid black;
}