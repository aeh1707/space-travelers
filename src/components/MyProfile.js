import { useSelector } from 'react-redux';

const MyProfile = () => {
  const missions = useSelector((state) => state.missions);
  const joMissions = missions.filter((mission) => mission.reserved);
  return (
    <>
      <h1>My Missions</h1>
      <ul>
        {joMissions.map((mission) => (<li key={mission.mission_id}>{mission.mission_name}</li>))}
      </ul>
    </>
  );
};
export default MyProfile;
