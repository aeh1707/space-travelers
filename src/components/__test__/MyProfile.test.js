import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import MyProfile from '../MyProfile';
import store from '../../redux/configureStore';

describe('Jest test', () => {
  it('Match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <MyProfile />
        </Provider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
