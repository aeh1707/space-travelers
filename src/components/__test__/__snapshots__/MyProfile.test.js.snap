// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest test Match snapshot 1`] = `
<h1>
  My Missions
</h1>
`;
